### PR TITLE
Transaction Details → Dispute Details – show dispute amount in store/dispute currency

### LINKED ISSUE #7124: Transaction details → use store currency as headline transaction amount
Part of #6883

On the **Transaction → Details** screen, we want to change the headline transaction amount to display in the merchant's default/primary currency, leaving the transaction currency as a detail within the breakdown.

This applies to stores that have
1. enabled multi-currency
1. a single deposit currency

transaction headline currency

Relevant discussion about closing this issue here: p1695933636976819-slack-C02BW3Z8SHK

**PRs**
- [ ] #7159 
- [x] #7287

### PR DESCRIPTION
**Pull request by Jinksi**
Part of #7124

> **Note**
> This PR is behind feature flag `_wcpay_feature_dispute_on_transaction_page`

#### Changes proposed in this Pull Request
This PR changes the Transaction Details → dispute amount to display in the store's currency rather than the charge currency, matching the currency of the headline transaction amount (changed in #7159).

For inquiries (dispute status `warning_needs_response`), the dispute amount will be rendered in the charge currency due to limitations with the dispute API – since no dispute balance transaction has been deducted, the amount is provided only in the charge currency.

[See design](https://www.figma.com/file/RiEQmaKRI7u9USAI3lyZz0VX/Dispute-Flows?node-id=6988:122148&mode=dev)

**Dispute** amount rendered in store currency

dispute

**Inquiry** dispute amount rendered in charge currency

inquiry

#### Testing instructions

1. Enable feature flag by running `update_option( '_wcpay_feature_dispute_on_transaction_page', '1' );` in [WP Console](https://wordpress.org/plugins/wp-console/) or by modifying your database table `wp_options` directly.
1. Enable **Payments → Settings → Multi-currency**.
3. Visit **Payments → Overview** and note your store's deposit currency or currencies, visible as balance tabs.
4. Enable a non-store currency in **WooCommerce → Settings → Multi-currency**. Enable the currency switcher if using the Storefront theme (otherwise, show the front-end currency switcher via the block editor or another method). 
6. Create a **disputed transaction in (one of) the store's currency** using the card `4000000000000259` at checkout. The transaction details screen should show the dispute amount in the store's currency, ~matching the headline transaction currency~.
8. Create a **disputed transaction in a non-store currency** using the currency switcher and the card `4000000000000259` at checkout. The transaction details screen should show the dispute amount in the store's currency, ~matching the headline transaction currency~.
8. Create a **disputed inquiry in a non-store currency** using the currency switcher and the card `4000000000001976` at checkout. The transaction details screen should show the dispute amount in the transaction currency.

**Update to test instructions**
* I've changes the base to `develop`, therefore the transaction headline amount will not display in the store's currency until #7159 is merged.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7287` or branch name `update/7124-transaction-details-dispute-amount-currency` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=update/7124-transaction-details-dispute-amount-currency) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 09ad2a77bc74283ff760b714a7624fb1b4a69b0e
- Build time: 2023-09-27 22:20:06 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Comment by haszari** [comment]
> This PR is based on the branch from PR https://github.com/Automattic/woocommerce-payments/pull/7159.

Curious – why is this PR coupled / dependent on the other one? Should I review differently?

File: client/disputes/utils.ts [hunk 1]
```diff
@@ -76,7 +76,7 @@ export const isInquiry = ( dispute: Pick< Dispute, 'status' > ): boolean => {
  * Returns the dispute fee balance transaction for a dispute if it exists
  * and the deduction has not been reversed.
  */
-const getDisputeDeductedBalanceTransaction = (
+export const getDisputeDeductedBalanceTransaction = (
```
Comment (haszari) [inline client/disputes/utils.ts:79]: Interesting … was this not accessible before?

**Review by haszari — APPROVED** [review]
Code looks fine and I did not see any issues† with this in my testing.

- I tested a range of existing multi-currency orders with disputes / inquiries.
- I also disabled multi-currency and tested a new disputed order. This showed the correct values too.

---

†I saw one bug we might want to log as a follow up. When a store disables multi-currency, previous (multicurrency) transactions don't show the relevant (historic) currency symbols, which could be confusing. 

E.g. in this case, the purchase was in NZD, but the transaction detail page doesn't make that clear.
Screenshot 2023-09-27 at 2 36 01 PM

This behaviour is also on `develop` (and likely has been for some time). Also relatively low priority, as most stores would not toggle mutlicurrency on and off.

File: client/disputes/utils.ts (hunk 1 above)
Comment (Jinksi) [reply client/disputes/utils.ts:79]: It was no longer used outside of this file once the "wrapper" function to get the formatted fee was introduced and used instead.

But now I have a use-case for it again.

**Comment by Jinksi** [comment]
> Curious – why is this PR coupled / dependent on the other one? Should I review differently?

@haszari This is a good point, there was no need to base this on the other PR apart from some convenience (shared configuration in test files).

I've changed the base to `develop` to allow this to be addressed/merged in isolation.

**Pull request merged**